src/services/WarehouseCrawlDataServices.ts

```diff
@@ -0,0 +1,84 @@
+import puppeteer from 'puppeteer';
+import winston from '../config/winston';
+import request_promise from 'request-promise';
+import cheerio from 'cheerio';
+
+async function WarehouseCrawlDataServices() {
+  try {
+    const browser = await puppeteer.launch({ headless: false });
+    const page = await browser.newPage();
+    await page.setRequestInterception(true);
+    await page.on('request', (req) => {
+      const type = ['image', 'media'];
+      const videos = [ '.mp4', '.avi', '.flv', '.mov', '.wmv'];
+      const url = req.url().toLowerCase();
+      const resourceType = req.resourceType();
+      const total = url.split('.').length - 1;
+      const video = url.split('\'')[total];
```

Comment:
đặt biến nó rõ nghĩa hơn

type của cái gì ?
videos => videoExtension 
....

Reply:
em đã commit nhưng chỗ cần chú ý và sủa type => typeImageVideo and videos=> videoExtension and video => videoTail